File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/data/model/UIState.java
package com.enpassio.twowaydatabinding.data.model;

//This enum class is for facilitating the switch between UI state
public enum UIState{

    //While user wait for data to be fetched, a loading indicator is shown
    LOADING,

    //When there is data to show, list can become visible
    HAS_DATA,

    //When there is no data in the inventory, empty image will be shown
    EMPTY
}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/data/UIState.kt
package com.enpassion.twowaydatabindingkotlin.data

//This enum class is for facilitating the switch between UI state
enum class UIState {
    //While user wait for data to be fetched, a loading indicator is shown
    LOADING,

    //When there is data to show, list can become visible
    HAS_DATA,

    //When there is no data in the inventory, empty image will be shown
    EMPTY
}